File: Lecture005 Bitwise Operators, For Loops, Operator Precedence & Variable Scoping/forPrime.cpp
#include<iostream>
using namespace std;

int main() {

    int n ;
    cout<<" enter the value of n" <<endl;
    cin >> n;

    bool isPrime = 1 ;

    for(int i = 2; i<n; i++) {

        //rem = 0, Not a Prime
        if(n%i == 0) {
            //cout<<" Not a Prime Number" << endl;
            isPrime = 0;
            break;
        }
    }

    if(isPrime == 0) {
        cout<<" Not a Prime Number"<<endl;
    }
    else
    {
        cout<<"is a Prime Number "<<endl;
    }

    return 0;
}

File: Lecture005 Bitwise Operators, For Loops, Operator Precedence & Variable Scoping/forPrime.java
import java.util.Scanner;

class forPrime {
    public static void main(String[] args) {
        int n;

        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the value of n: ");
        n = sc.nextInt();
        sc.close();

        boolean isPrime = true;

        for (int i = 2; i < n; i++) {
            // rem is 0, not a prime
            if (n % i == 0) {
                isPrime = false;
                break;
            }
        }

        if (isPrime) {
            System.out.println("is a prime number.");
        } else {
            System.out.println("not a prime number.");
        }
    }
}
